app/controllers/tasks_controller.rb

```diff
@@ -208,6 +210,12 @@ def only_authorize_action
     authorize Task
   end
 
+  def redirect_task_contribution
+    return unless @task.present? && @task.contribution?
+
+    redirect_to action: action_name, controller: 'task_contributions', task_id: @task.parent, id: @task.task_contribution
```

Comment:
I just got an `ActionController::UrlGenerationError` here. However, I am not sure whether this is some corruption in my local test data or a real issue that could re-occur.

<details>
<summary>Local Reproduction</summary>

I tested with <http://localhost:7500/tasks/856>, which is a contribution belonging to a task that was deleted (probably belonging to <http://localhost:7500/tasks/24>, deleted on `2024-07-09 23:38:44 +0200`.

</details>

Maybe we still ensure to catch those errors?

Reply:
It is an actual issue, which is caused by the fact that deleting a task does not yet modify its associated contributions.

[See this comment for reference](https://github.com/openHPI/codeharbor/pull/1103#discussion_r1671273478)

Reply:
Okay, thanks. One could still argue to add a rescue here, but I would leave it in the current state for now. Thus resolving this thread.